pkg/controller/address_space_controller/controller.go

```diff
@@ -212,19 +212,24 @@ func ApplyDeployment(deployment *appsv1.Deployment) error {
 			deployment.ObjectMeta.Annotations["prometheus.io/port"] = "8080"
 		}
 
-		memoryEnv := util.GetEnvOrDefault("ADDRESS_SPACE_CONTROLLER_MEMORY_LIMIT", "512Mi")
-		memoryLimit, err := resource.ParseQuantity(memoryEnv)
-		if err != nil {
-			return err
+		cpuEnv, ok := os.LookupEnv("ADDRESS_SPACE_CONTROLLER_CPU_LIMIT")
+		if ok {
+			cpuLimit, err := resource.ParseQuantity(cpuEnv)
+			if err != nil {
+				return err
+			}
+			container.Resources.Requests[corev1.ResourceCPU] = cpuLimit
+			container.Resources.Limits[corev1.ResourceCPU] = cpuLimit
```

Comment:
if the user removes the ADDRESS_SPACE_CONTROLLER_CPU_LIMIT var, I think the existing container would still contain the stale values.  we should remove them on the false path.

Reply:
Fixed.